packages/go_router_builder/lib/src/go_router_generator.dart

```diff
@@ -61,48 +62,53 @@ ${getters.map((String e) => "$e,").join('\n')}
     Set<String> values,
     Set<String> getters,
   ) {
-    for (final AnnotatedElement annotatedElement
-        in library.annotatedWith(_typeChecker)) {
+    for (final AnnotatedElement annotatedElement in library.annotatedWith(
+      _typeChecker,
+    )) {
       final InfoIterable generatedValue = _generateForAnnotatedElement(
         annotatedElement.element,
         annotatedElement.annotation,
       );
       getters.add(generatedValue.routeGetterName);
-      for (final String value in generatedValue) {
-        assert(value.length == value.trim().length);
-        values.add(value);
-      }
+      values.addAll(generatedValue.members);
     }
   }
 
   InfoIterable _generateForAnnotatedElement(
-    Element element,
+    Element2 element,
```

Comment:
looking at the doc https://pub.dev/documentation/analyzer/latest/dart_element_element/Element2.html

should we keep the original class name

Reply:
@chunhtai 
Element2 will be deprecated starting with analyzer v8.0.0. Since `analyzer: “>=7.4.0 <8.0.0”` is set, Element2 is not deprecated.

https://pub.dev/documentation/analyzer/7.7.1/dart_element_element2/Element2-class.html
https://pub.dev/documentation/analyzer/8.0.0/dart_element_element/Element2.html

Other libraries are also replacing Element2 in the same way.

https://github.com/google/json_serializable.dart/pull/1504
https://github.com/rrousselGit/freezed/pull/1276

Reply:
can we ping analyzer 8.0.0 directly?

Reply:
@chunhtai 
Specifying `^8.0.0` will break compatibility with other packages.

https://github.com/dart-lang/mockito/blob/v5.4.5/pubspec.yaml
https://github.com/google/json_serializable.dart/blob/json_serializable-v6.10.0/json_serializable/pubspec.yaml
https://github.com/rrousselGit/freezed/blob/freezed-v3.2.0/packages/freezed/pubspec.yaml